File: 2 Array:List/3.passByRef.py
def changeArr(arr):
    # Pass by reference (mutable object - list)
    for i in range(len(arr)):
        arr[i] = 2 * arr[i]

arr = [1, 2, 3]
changeArr(arr)  # Modifies the original list
for num in arr:
    print(num)

File: 2 Array:List/3.pass_by_ref.js
function changeArr(arr) {
    // Pass by reference (mutable object - array)
    for (let i = 0; i < arr.length; i++) {
        arr[i] = 2 * arr[i];
    }
}

let arr = [1, 2, 3];
changeArr(arr);  // Modifies the original array
arr.forEach(num => console.log(num));
